components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/ApiStoreSdkGenerationManager.java

```diff
@@ -0,0 +1,226 @@
+        if (api == null) {
+            String errorMessage = "Cannot find API for specified API ID";
+            throw new APIManagementException(errorMessage, ExceptionCodes.API_NOT_FOUND);
+        }
+        String apiName = api.getName();
+        String apiVersion = api.getVersion();
+        String swaggerDefinitionForApi = null;
+        try {
+            swaggerDefinitionForApi = apiStore.getApiSwaggerDefinition(apiId);
+        } catch (APIManagementException e) {
+            handleSdkGenException("Error retrieving swagger definition for API " + apiId + " from database.", e);
+        }
+        Swagger swaggerDoc = new SwaggerParser().parse(swaggerDefinitionForApi);
+        if (swaggerDoc == null) {
+            handleSdkGenException("Error while parsing retrieved swagger definition");
+        }
+        //Format the swagger definition as a string before writing to the file.
+        String formattedSwaggerDefinitionForSdk = Json.pretty(swaggerDoc);
+        Path tempSdkGenDir = null;
+        File swaggerDefJsonFile = null;
+        try {
+            //Create a temporary directory to store the API files
+            tempSdkGenDir = Files.createTempDirectory(apiName + "_" + language + "_" + apiVersion);
+
+            //Create a temporary file to store the swagger definition
+            swaggerDefJsonFile = Files.createTempFile(tempSdkGenDir,
+                    apiId + "_" + language,
+                    APIMgtConstants.APIFileUtilConstants.JSON_EXTENSION).toFile();
+        } catch (IOException e) {
+            handleSdkGenException("Error creating temporary directory or json file for swagger definition!", e);
+        }
+
+        String tempZipFilePath = "";
+        if (swaggerDefJsonFile.exists()) {
+
+            try (Writer swaggerFileWriter = new BufferedWriter(
+                    new OutputStreamWriter(
+                            new FileOutputStream(swaggerDefJsonFile.getAbsoluteFile()), "UTF-8"))) {
+
+                swaggerFileWriter.write(formattedSwaggerDefinitionForSdk);
+
+                if (log.isDebugEnabled()) {
+                    log.debug("Writing the swagger definition was sucessful to file " +
+                            swaggerDefJsonFile.getAbsolutePath());
+                }
+            } catch (IOException e) {
+                handleSdkGenException("Error writing swagger definition to file in " +
+                        tempSdkGenDir, e);
+            }
+
+            //Generate the SDK for the specified language
+            generateSdkForSwaggerDef(language, swaggerDefJsonFile.getAbsolutePath(), tempSdkGenDir.toString());
+
+            String archiveName = apiName + "_" + language + "_" + apiVersion;
+            tempZipFilePath = tempSdkGenDir + "/" + archiveName + ".zip";
+            APIFileUtils.archiveDirectory(tempSdkGenDir.toString(),
+                    tempSdkGenDir.toString(),
+                    archiveName);
+            if (log.isDebugEnabled()) {
```

Comment:
Now we are not using is debug enable. Instead using singline debug statements with templates can we change this?